### PR TITLE
Fix drawers are draggable on desktop platforms

### LINKED ISSUE #100321: It's possbile to drag Drawers with the mouse
Currently, Drawers can be dragged open and closed with the mouse:

https://user-images.githubusercontent.com/389558/158868272-c2aacf4e-1c1c-404c-9b45-004594f6d066.mp4

There is no way to disable this behavior.  We should either provide a way to disable this, or if no one wants this behavior at all, remove it completely for mouse interactions.

### To reproduce

  1. Run an app with a Drawer, such as the one given below.
  2. Make sure you have a mouse connected.
  3. Drag from the left side of the screen with the mouse.

Expected: Nothing happens
Actual: The menu is revealed.  It can also be dragged shut.

<details>

<summary>Code</summary>

```dart
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        platform: TargetPlatform.macOS,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Drawer Demo'),
        ),
        drawer: Drawer(
          child: ListView(
            padding: EdgeInsets.zero,
            children: const <Widget>[
              DrawerHeader(
                decoration: BoxDecoration(
                  color: Colors.blue,
                ),
                child: Text(
                  'Drawer Header',
                  style: TextStyle(
                    color: Colors.white,
                    fontSize: 24,
                  ),
                ),
              ),
              ListTile(
                leading: Icon(Icons.message),
                title: Text('Messages'),
              ),
              ListTile(
                leading: Icon(Icons.account_circle),
                title: Text('Profile'),
              ),
              ListTile(
                leading: Icon(Icons.settings),
                title: Text('Settings'),
              ),
            ],
          ),
        ),
        body: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            CupertinoTextField(
              maxLines: 3,
            ),
            SizedBox(height: 32.0),
            TextField(
              maxLines: 3,
              decoration: InputDecoration(
                border: OutlineInputBorder(
                  borderRadius: BorderRadius.circular(30.0),
                  gapPadding: 0.0,
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```

</details>

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/100321

<details> 
<summary>minimal code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({Key? key, this.dark = true}) : super(key: key);

  final bool dark;

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      title: 'Material App',
      theme: dark ? ThemeData.dark() : ThemeData.light(),
      home: const Home(),
    );
  }
}

class Home extends StatelessWidget {
  const Home({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Material App Bar'),
      ),
      drawer: const Drawer(),
      endDrawer: const Drawer(),
    );
  }
}
``` 
	
</details>

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/100476

**Review by justinmc — COMMENTED** [review]
What about mobile platforms with a mouse connected or desktop platforms with a touchscreen?

Maybe in this case it's not so easy to handle those cases and we should just go with this for now...  This mouse/touch and desktop/mobile distinction is something we haven't done a good job being consistent about in the history of the framework and we're going to have to figure it out at some point.

File: packages/flutter/lib/src/material/drawer.dart [hunk 1]
```diff
@@ -553,6 +553,19 @@ class DrawerControllerState extends State<DrawerController> with SingleTickerPro
...
+        case TargetPlatform.iOS:
+        case TargetPlatform.fuchsia:
+          isDesktop = false;
+          break;
+        case TargetPlatform.macOS:
+        case TargetPlatform.linux:
+        case TargetPlatform.windows:
+          isDesktop = true;
+          break;
+      }
```
[the remark this answers is not in the document]
Comment (TahaTesser) [reply packages/flutter/lib/src/material/drawer.dart]: Thanks for the feedback, apologies for the delay, looks like this PR fell off my radar (including couple).

> Nit: Should this be indented like this? I haven't seen it before in the framework but kind of makes sense...


Fixed.

> Maybe in this case it's not so easy to handle those cases and we should just go with this for now... This mouse/touch and desktop/mobile distinction is something we haven't done a good job being consistent about in the history of the framework and we're going to have to figure it out at some point.

If you an file an issue, please tag me, I'd love to help in any way :)

**Review by justinmc — APPROVED** [review]
LGTM :+1: 

I'll plan to gather a few examples of the desktop vs. mobile-with-keyboard-and-mouse distinction and then open an issue.

**Comment by talamaska** [comment]
I think this should be reverted. I don't like when the framework is making decisions on my behalf. And this change makes the 2 settings for disabling the drag obsolete on desktop, which is bad. If the dev wants to block the drag, he/she should use the existing props to do it depending on the platform.  [drawerEnableOpenDragGesture](https://master-api.flutter.dev/flutter/material/Scaffold/drawerEnableOpenDragGesture.html) and [endDrawerEnableOpenDragGesture](https://master-api.flutter.dev/flutter/material/Scaffold/endDrawerEnableOpenDragGesture.html) allows to disable this

**Comment by talamaska** [comment]
Are you saying that desktop doesn't have drag and drop gesture? 
Then what is  [this](https://www.w3schools.com/html/html5_draganddrop.asp) ?
Scroll list are something completely different.
I'm saying that you are making a global change of the framework because of one person and one issue with zero thumbs up reactions.

**Pull request merged**